src/Dataset.ts

```diff
@@ -15,11 +15,12 @@ type Key = string;
 
 export abstract class Dataset<T extends Tile> {
   abstract root_tile : T;
-//  public mutations : 
   public max_ix : number = -1;
   protected plot : Scatterplot;
   protected _tileworkers: TileWorker[] = [];
   abstract ready : Promise<void>;
+  abstract download_most_needed_tiles(bbox : Rectangle, max_ix: number, queue_length : number) : void;
```

Comment:
Consider placing abstract methods below the constructor (with the other methods) for clarity